lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -85,9 +85,11 @@ static bool isPort(Value value) {
 }
 
 /// Gets the port for a given BlockArgument.
-ComponentPortInfo calyx::getComponentPortInfo(BlockArgument arg) {
+PortInfo calyx::getPortInfo(BlockArgument arg) {
   Operation *op = arg.getOwner()->getParentOp();
-  return getComponentPortInfo(op)[arg.getArgNumber()];
+  assert(isa<ComponentOp>(op) &&
+         "Only ComponentOp should support lookup by BlockArgument.");
+  return cast<ComponentOp>(op).getPortInfo()[arg.getArgNumber()];
```

Comment:
`getInputPortInfo`?

Reply:
Not necessarily, both input and output ports are passed as `BlockArgument`s. Good point about the repetition. Maybe some kind of share-able interface would be good here.

Reply:
Hmm on second thought, the idea of always casting to some `HasPortsInterface` seems kind of tedious. However, we should share logic/names for each.

Reply:
I'm going to push this for now, and come back to refactoring the two. I think the most important objective of this is using `PortInfo`.